src/bodies.cpp

```diff
@@ -699,9 +711,16 @@ bool bodies::Box::intersectsRay(const Eigen::Vector3d& origin, const Eigen::Vect
   {
     if (tmax - tmin > detail::ZERO)
     {
-      intersections->push_back(tmin * dir + origin);
-      if (count > 1)
+      if (tmin > detail::ZERO)
+      {
+        intersections->push_back(tmin * dir + origin);
+        if (count > 1)
+          intersections->push_back(tmax * dir + origin);
+      }
+      else
+      {
         intersections->push_back(tmax * dir + origin);
```

Comment:
Does `tmin <= detail::ZERO` really imply origin is inside the box?
If it does, please add a comment to make this clear here.

Reply:
That's it. I added comments to the code which should explain everything.

Reply:
wonderful, thanks! :sun_with_face: